packages/scandipwa/src/component/SearchOverlay/SearchOverlay.component.tsx

```diff
@@ -80,22 +88,51 @@ export class SearchOverlayComponent extends PureComponent<SearchOverlayComponent
         );
     }
 
+    renderSearchOverlayResults(): ReactElement {
+        const { activeClosingAnimation } = this.state;
+
+        return (
+            <div
+              block="SearchOverlay"
+              elem="Results"
+              aria-label="Search results"
+              mods={ { activeClosingAnimation } }
+              ref={ this.resultRef }
+            >
+                { this.renderSearchResults() }
+            </div>
+        );
+    }
+
     render(): ReactElement {
         const { isHideOverlay, searchCriteria } = this.props;
+        const { prevSearchCriteria, activeClosingAnimation } = this.state;
+
+        if (!searchCriteria.trim() && searchCriteria !== prevSearchCriteria) {
+            this.setState({ activeClosingAnimation: true });
+
+            const animationendHandler = () => {
+                this.setState({ activeClosingAnimation: false });
+
+                this.resultRef.current?.removeEventListener('animationend', animationendHandler);
+            };
 
-        if (!searchCriteria.trim()) {
+            this.resultRef.current?.addEventListener('animationend', animationendHandler);
+        } else if (!searchCriteria.trim() && !activeClosingAnimation) {
             return null;
         }
```

Comment:
no `setState` calls or event listener attachments in rendering methods, refactor the implementation by moving this to lifecycle methods, ideally to the container ( `SearchOverlay.container` ).